theworkoutinfo/ViewControllers/HomeViewController.swift

```diff
@@ -57,22 +67,51 @@ class HomeViewController: UIViewController {
         self.tableView.es.addInfiniteScrolling {
             [unowned self] in
             self.pageNumber += 1
-            self.viewModel.loadExercises.onNext(self.pageNumber)
+            self.isCategorySelected ? self.viewModel.loadExercisesByCategory.onNext((false, self.categorySelected!, self.pageNumber)) : self.viewModel.loadExercises.onNext((false, self.pageNumber))
         }
+        
+        categoryTextField.inputView = categoryPickerView
+        categoryTextField.text = "Select a category"
+        filterView.backgroundColor = UIColor.greenAppColor()
     }
     
     func setupBindings() {
+        viewModel.loadExercises.onNext((false,1))
+        viewModel.loadCategories.onNext(())
+        
         viewModel.exercises.asObservable().subscribe(onNext: { _ in
             self.rc.endRefreshing()
             self.tableView.es.stopLoadingMore()
-            print("It Worked!")
         }).disposed(by: disposeBag)
         
         rc.rx.controlEvent(.valueChanged).subscribe({ _ in
-            self.viewModel.loadExercises.onNext(1)
+            self.pageNumber = 1
+            self.isCategorySelected ? self.viewModel.loadExercisesByCategory.onNext((false, self.categorySelected! ,self.pageNumber)) : self.viewModel.loadExercises.onNext((false, 1))
         }).disposed(by: disposeBag)
         
-        self.viewModel.loadExercises.onNext(pageNumber)
+        viewModel.categories.asObservable().bind(to: categoryPickerView.rx.itemTitles) { _, item in
+            return item.name
+            }.disposed(by: disposeBag)
+        
+        categoryPickerView.rx.modelSelected(ExerciseCategory.self)
+            .subscribe(onNext: {
+                categoriesSelected in
+                
+                guard let category = categoriesSelected.first else {
+                    return
+                }
+                
+                self.categoryTextField.text = category.name
+                self.categorySelected = category
```

Comment:
All these changes need to happen after a category is selected, not after the category is selected in the picker.
I mean, the picker sects a category and we map that to the selectedcategory observable in the viewmodel. Then in the ViewController we listen to that category and setup the UI accordingly, but not from the picker selection.

Also keep in mind that somee of the stuff like the page number could have to be done in the Viewmodel instead of the VC. Remember that the viewmodel is the one that defines which data is shown and the VC only presents it

Reply:
Most of what i see below and that reloads data based on the new category should be done in the vm from what i see